src/features/mcda/atoms/mcdaCalculation.ts

```diff
@@ -0,0 +1,223 @@
+import isEqual from 'lodash/isEqual';
+import sum from 'lodash/sum';
+import { createAtom } from '~utils/atoms';
+import { layersRegistryAtom } from '~core/logical_layers/atoms/layersRegistry';
+import { layersSettingsAtom } from '~core/logical_layers/atoms/layersSettings';
+import { createUpdateLayerActions } from '~core/logical_layers/utils/createUpdateActions';
+import {
+  anyCondition,
+  featureProp,
+  notEqual,
+} from '~utils/bivariate/bivariate_style/styleGen';
+import { adaptTileUrl } from '~utils/bivariate/tile/adaptTileUrl';
+import appConfig from '~core/app_config';
+import {
+  MCDARenderer,
+  sentimentDefault,
+  sentimentReversed,
+} from '../renderers/MCDARenderer';
+import type { BivariateLayerStyle } from '~utils/bivariate/bivariateColorThemeUtils';
+import type { Quotient } from '~utils/bivariate';
+
+export interface JsonMCDA {
+  id?: string;
+  axes: [string, string][];
+  ranges: [number, number][];
+  sentiments: [string, string][];
+  coefficients: number[];
+  colors: {
+    good: string;
+    bad: string;
+  };
+}
+
+const DEFAULT_GREEN = 'rgba(90, 200, 127, 0.5)';
+const DEFAULT_RED = 'rgba(228, 26, 28, 0.5)';
+
+const filterSetup = (quotients: Quotient[]) =>
+  anyCondition(
+    ...quotients.map((quotient) =>
+      notEqual(['/', featureProp(quotient[0]), featureProp(quotient[1])], 0),
+    ),
+  );
```

Comment:
So what's happening here? You filter out quotients that are not equal to '/'? Or quotiens whos division not equal to 0?

Upd. i see. It creates some sort of mapbox style rules...

Reply:
yeah, it's a copy of filterSetup from src/utils/bivariate/bivariate_style/index.ts
so we filter only those num/den !== 0